datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -229,6 +245,7 @@ protected void restartDeadTasks() {
         _logger.warn("Creating a new connector task for the datastream task {}", datastreamTask);
         _runningTasks.put(datastreamTask, createKafkaConnectorTask(datastreamTask));
       });
+      _dynamicMetricsManager.createOrUpdateCounter(_metricsPrefix, NUM_TASK_RESTARTS, deadDatastreamTasks.size());
```

Comment:
Sonam left me a question about counters vs. gauges but GitHub won't allow me to comment on it for some reason — they're roughly comparable but [this](https://iwww.corp.linkedin.com/wiki/cf/display/ENGS/Health+Check+Sensor+Basics#HealthCheckSensorBasics-CountervsGauge) ingraphs wiki page seems to suggest we should be using gauges for values that would increase and decrease over time (which seems to be the case for the metric this PR is adding). A similar [comment](https://github.com/dropwizard/metrics/issues/1043#issuecomment-272706512) has been made in response to the same question on the repo of the metrics library we're using.

Reply:
As discussed made this a Gauge.